File: arrays_strings/three.go
package main

import "fmt"
import "github.com/golang-collections/go-datastructures/bitarray"

func main() {
	fmt.Println("is palindrome permutation")
	fmt.Println(tester("tacocat", true))
	fmt.Println(tester("abcab", true))
	fmt.Println(tester("abz", false))
}

func solution(input string) bool {
	bitVector := bitarray.NewBitArray(27)
	for _, val := range input {
		i := uint64(val - 97)
		bit, _ := bitVector.GetBit(i)
		if bit {
			bitVector.ClearBit(i)
		} else {
			bitVector.SetBit(i)
		}
	}
	sol := bitVector.ToNums()
	return len(sol) <= 1
}

func tester(input string, expected bool) string {
	if solution(input) == expected {
		return "pass"
	}
	return "fail"
}

File: arrays_strings/three.js
console.log("palindrome permutation");

test("tactcoa", true);
test("eat", false);
test("abab", true);

function sol(a) {
    const sorted = [...a].sort();
    let odds = 0;
    let counter = 1;
    for (let i=1;i<sorted.length;i++) {
        if (sorted[i-1] === sorted[i]) {
            counter++;
        } else {
            if (counter == 1 || counter%2 != 0) {
                odds++
            }
            if (odds > 1) {
                return false;
            }
            counter = 1;
        }
    }
    return true;
}

function test(val, expected) {
    if (sol(val) == expected) {
        console.log("pass");
        return;
    }
    console.log("fail");
}